src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,158 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.BlockSyncer
+  (
+    blockSyncerBracket
+  , runBlockLayerMockup
+  , BlocksConsumed(..)
+  ) where
+
+import Prelude
+
+
+import Control.Concurrent
+    ( forkIO, killThread, myThreadId, threadDelay )
+import Control.Exception
+    ( SomeException, bracket, catch )
+import Data.IORef
+    ( IORef, newIORef, readIORef, writeIORef )
+import Data.List
+    ( delete, (!!) )
+import Data.Text.Internal
+    ( showText )
+import Data.Word
+    ( Word64 )
+import Formatting
+    ( sformat, (%) )
+import qualified Formatting as F
+
+
+import Cardano.Wallet.Primitive
+    ( Block (..), BlockHeader (..), Hash (..) )
+
+
+
+newtype BlocksConsumed = BlocksConsumed [(Hash "BlockHeader")] deriving (Show, Eq)
+
+data BlockLayerMockup = BlockLayerMockup {
+      getBlock      :: (Hash "BlockHeader") -> IO Block
+    , getEpoch      :: Word64 -> IO [Block]
+    , getNetworkTip :: IO ((Hash "BlockHeader"),BlockHeader)
+    }
```

Comment:
This, and the implementation below of the mockup are only for testing I presume :thinking: ? I wouldn't expect this to be needed for writing and testing the ticking function. Cf next comment.

Reply:
it was supposed to act like that. in the light of the refactoring it is not used anymore